pgdb/v1/fts.go

```diff
@@ -129,22 +112,117 @@ func lemmatizeDocs(docs []*SearchContent, additionalFilters ...jargon.Filter) []
 	return rv
 }
 
+func camelSplitDoc(docValue string, doc *SearchContent) []lexeme {
+	rv := make([]lexeme, 0, 8)
+	var buffer bytes.Buffer
+	var pos = 1
+	var prev rune
+	for _, r := range docValue {
+		if prev == 0 {
+			prev = r
+			continue
+		}
+		if unicode.IsUpper(prev) {
+			if buffer.Len() == 0 { // no current word
+				if unicode.IsLower(r) {
+					// got a upper case in prev and current is lower, starting a new word
+					if _, e := buffer.WriteRune(prev); e != nil {
+						buffer = bytes.Buffer{}
```

Comment:
shouldn't this be buffer.Reset() so the underlying slice is reused?

Reply:
Oh right ec2a9f7d9a9bd0e6b7225ab81d52fd7784bd276c